uaclient/util.py

```diff
@@ -319,28 +319,35 @@ def parse_os_release(release_file: 'Optional[str]' = None) -> 'Dict[str, str]':
     return data
 
 
-REGEX_OS_RELEASE_VERSION_1 = (  # Precise, Trusty
-    r'(?P<version>\d+\.\d+)(\.\d)? (?P<lts>LTS,?) ?(?P<series>\w+).*')
-REGEX_OS_RELEASE_VERSION_2 = (  # >= Disco
-    r'(?P<version>\d+\.\d+)(\.\d)? (?P<lts>LTS )?\((?P<series>\w+).*')
+# N.B. this relies on the version normalisation we perform in get_platform_info
+REGEX_OS_RELEASE_VERSION = r'(?P<release>\d+\.\d+) (LTS )?\((?P<series>\w+).*'
 
 
 def get_platform_info() -> 'Dict[str, str]':
+    """
+    Returns a dict of platform information.
+
+    N.B. This dict is sent to the contract server, which requires the
+    distribution, type and release keys.
+    """
     os_release = parse_os_release()
     platform_info = {
         'distribution': os_release.get('NAME', 'UNKNOWN'),
         'type': 'Linux'}
 
     version = os_release['VERSION']
-    match = re.match(REGEX_OS_RELEASE_VERSION_1, version)
-    if not match:
-        match = re.match(REGEX_OS_RELEASE_VERSION_2, version)
+    if ', ' in version:
+        version = '{} ({})'.format(*version.split(', '))
+    version = re.sub(r'\.\d LTS', ' LTS', version)
```

Comment:
Do we need a '+'        r'\.\d+ LTS'   ?

Reply:
We don't, this is just stripping off the point release, so there should be exactly one digit. (I don't know what pre-point-release LTS's look like, but if they are "20.04 LTS" then this would be fine but \d+ would give us "20 LTS", I think.)

I'll add a comment to make what this is doing clearer.